src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -132,65 +132,49 @@ public async Task SslStream_StreamToStream_Alpn_Success(List<SslApplicationProto
-                    using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+                // Test ALPN failure only on platforms that supports ALPN.
+                if (BackendSupportsAlpn)
+                {
+                    Task t1 = Assert.ThrowsAsync<AuthenticationException>(() => clientStream.AuthenticateAsClientAsync(TestAuthenticateAsync, clientOptions));
+
+                    try
                     {
-                        SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions
-                        {
-                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http2 },
-                            ServerCertificate = certificate,
-                        };
-                        SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
-                        {
-                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http11 },
-                            RemoteCertificateValidationCallback = AllowAnyServerCertificate,
-                            TargetHost = certificate.GetNameInfo(X509NameType.SimpleName, false),
-                        };
+                        await serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions);
+                        Assert.True(false, "AuthenticationException was not thrown.");
+                    }
+                    catch (AuthenticationException) { serverStream.Dispose(); }
 
-                        // Test alpn failure only on platforms that supports ALPN.
-                        if (BackendSupportsAlpn)
-                        {
-                            // schannel sends alert on ALPN failure, openssl does not.
-                            Task t1 = Assert.ThrowsAsync(TestConfiguration.SupportsAlpnAlerts ? typeof(AuthenticationException) : typeof(IOException), () =>
-                                clientStream.AuthenticateAsClientAsync(TestAuthenticateAsync, clientOptions));
-
-                            try
-                            {
-                                await serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions);
-                                Assert.True(false, "AuthenticationException was not thrown.");
-                            }
-                            catch (AuthenticationException) { server.Dispose(); }
-
-                            await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1);
-                        }
-                        else
-                        {
-                            Task t1 = clientStream.AuthenticateAsClientAsync(TestAuthenticateAsync, clientOptions);
-                            Task t2 = serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions);
+                    await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1);
```

Comment:
WhenAllOrAnyFailedWithTimeout makes little sense when passed a single task :)

Reply:
yes. that was blind copy from the old test.

Reply:
I could not find WithTimeout, what do I miss? I saw proposal but that did not look like ready.

Reply:
In this library the test helper is actually called TimeoutAfter.

Reply:
updated. Would it make sense to make it consistent with other assemblies? (or wait for  #27723) ?

Reply:
> Would it make sense to make it consistent with other assemblies?

Yup